masterplan_tools/method/blocks/blocks_clustering.py

```diff
@@ -0,0 +1,415 @@
+from warnings import simplefilter
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from shapely.geometry import LineString, MultiPoint, MultiPolygon, Point, Polygon
+from shapely.ops import nearest_points
+from sklearn.cluster import DBSCAN
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.metrics import silhouette_score
+from tqdm import tqdm
+from .utils import Utils
+
+simplefilter("ignore", category=ConvergenceWarning)
+simplefilter(action="ignore", category=FutureWarning)
```

Comment:
Maybe it is better to write a code that does not generate those warnings instead of ignoring them? Especially by the `samplefilter` somewhere in the middle of the codebase where it would be hard to locate this statement.